pkg/certgenerator/v1beta1/generator_test.go

```diff
@@ -208,13 +210,43 @@ func TestGenerate(t *testing.T) {
 }
 
 func buildFakeClient(kubeResources []client.Object) client.Client {
-	fakeClientBuilder := fake.NewClientBuilder().WithScheme(scheme.Scheme)
+	fakeClientBuilder := fake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		WithInterceptorFuncs(interceptor.Funcs{Patch: ssaAsStrategicMergePatchFunc})
 	if len(kubeResources) > 0 {
 		fakeClientBuilder.WithObjects(kubeResources...)
 	}
 	return fakeClientBuilder.Build()
 }
 
+type ssaPatchAsStrategicMerge struct {
+	client.Patch
+}
+
+func (*ssaPatchAsStrategicMerge) Type() types.PatchType {
+	return types.StrategicMergePatchType
+}
+
+func wrapSSAPatch(patch client.Patch) client.Patch {
+	if patch.Type() == types.ApplyPatchType {
+		return &ssaPatchAsStrategicMerge{Patch: patch}
+	}
+	return patch
+}
+
+// ssaAsStrategicMergePatchFunc returns the patch interceptor.
+// TODO (tenzen-y): Once the fake client supports server-side apply, we should remove these interceptor.
+// REF:  https://github.com/kubernetes/kubernetes/issues/115598
+func ssaAsStrategicMergePatchFunc(
+	ctx context.Context,
+	cli client.WithWatch,
+	obj client.Object,
+	patch client.Patch,
+	opts ...client.PatchOption,
+) error {
+	return cli.Patch(ctx, obj, wrapSSAPatch(patch), opts...)
+}
```

Comment:
As I described in the comments, we need to keep this interceptor until the fake client supports SSA since the certgenerator uses the SSA here:

- https://github.com/kubeflow/katib/blob/1365e473c5bb438b80f95df0012a45579af4af25/pkg/certgenerator/v1beta1/generator.go#L258
- https://github.com/kubeflow/katib/blob/1365e473c5bb438b80f95df0012a45579af4af25/pkg/certgenerator/v1beta1/generator.go#L285

Reply:
Great, thanks for finding this!